dcd/usb_phydcd.c

```diff
@@ -100,154 +111,476 @@ usb_phydcd_status_t USB_PHYDCD_Control(usb_phydcd_handle handle, usb_phydcd_cont
-        case kUSB_DevicePHYDcdRun:
-            if (0U == dcdState->dcdDisable)
-            {
-                dcdState->dcdDetectState = (uint8_t)kUSB_DCDDetectStart;
-            }
-            break;
-        case kUSB_DevicePHYDcdStop:
-            if (0U == dcdState->dcdDisable)
-            {
-                dcdState->dcdDetectState = (uint8_t)kUSB_DCDDetectInit;
-            }
-            break;
-        case kUSB_DevicePHYDcdEnable:
-            dcdState->dcdDisable = 0U;
-            break;
-        case kUSB_DevicePHYDcdDisable:
-            dcdState->dcdDisable = 1U;
-            break;
-        default:
-            /*no action*/
-            break;
+    switch (type) {
+    case kUSB_DevicePHYDcdRun:
+        if (0U == dcdState->dcdDisable) {
+            dcdState->dcdDetectState = (uint8_t)kUSB_DCDDetectInit;
+        }
+        break;
+    case kUSB_DevicePHYDcdStop:
+        if (0U == dcdState->dcdDisable) {
+            dcdState->dcdDetectState = (uint8_t)kUSB_DCDDetectInit;
+        }
+        break;
+    case kUSB_DevicePHYDcdEnable:
+        dcdState->dcdDisable = 0U;
+        break;
+    case kUSB_DevicePHYDcdDisable:
+        dcdState->dcdDisable = 1U;
+        break;
+    default:
+        /*no action*/
+        break;
     }
     return dcdError;
 }
 
-usb_phydcd_status_t USB_PHYDCD_TimerIsrFunction(usb_phydcd_handle handle,const uint64_t tick)
+static void usbAnalogChargerDetectRegSet(usb_phydcd_state_struct_t *dcd, dcdBitState state, uint32_t mask)
 {
-    usb_phydcd_status_t dcdError = kStatus_phydcd_Success;
-    usb_phydcd_state_struct_t *dcdState;
-    dcdState = (usb_phydcd_state_struct_t *)handle;
-    USBPHY_Type *usbPhyBase;
-    usb_phydcd_dev_status_t dcdStatus;
-    if (NULL == handle)
-    {
-        return kStatus_phydcd_Error;
+    if (state == CLEAR) {
+        dcd->usbAnalogBase->INSTANCE[dcd->index].CHRG_DETECT_CLR = mask;
+    }
+    else {
+        dcd->usbAnalogBase->INSTANCE[dcd->index].CHRG_DETECT_SET = mask;
+    }
+}
+
+static void usbLoopbackRegSet(usb_phydcd_state_struct_t *dcd, dcdBitState state, uint32_t mask)
+{
+    if (state == CLEAR) {
+        dcd->usbAnalogBase->INSTANCE[dcd->index].LOOPBACK_CLR = mask;
+    }
+    else {
+        dcd->usbAnalogBase->INSTANCE[dcd->index].LOOPBACK_SET = mask;
+    }
+}
+
+static void usbPhyDebugRegSet(usb_phydcd_state_struct_t *dcd, dcdBitState state, uint32_t mask)
+{
+    if (state == CLEAR) {
+        ((USBPHY_Type *)dcd->phyBase)->DEBUG_CLR = mask;
+    }
+    else {
+        ((USBPHY_Type *)dcd->phyBase)->DEBUG_SET = mask;
+    }
+}
+
+static uint32_t usbAnalogChargerDetectStatGet(usb_phydcd_state_struct_t *dcd, uint32_t mask)
+{
+    return (dcd->usbAnalogBase->INSTANCE[dcd->index].CHRG_DETECT_STAT & mask);
+}
+
+static usb_phydcd_dev_status_t USB_PHYDCD_Apple_Check(usb_phydcd_state_struct_t *dcd)
+{
+    bool dp_state = false;
+    // bool dm_state = false;
+    LOG_INFO("Apple check");
+
+    if ((dcd->hwTick - dcd->startTime) >= 100U) {
```

Comment:
All magic numbers should be defined at the level or locally so that it’s easier to understand what they mean